2주차 미션 / 안드로이드 1조 정일혁

1hyok opened the pull request:
## 📝 미션
- [x] 미션 1
서울시 광진구 구의동 부분 컴포넌트
- [x] 미션 2
주문하러 가기 컴포넌트(주문하러 가기 부분 버튼으로 만들기)
- [x] 미션 3
최근 본 맛집의 음식점 부분 컴포넌트
- [x] 미션 4
돈가스, 일식 … 부분 LazyColumn으로 두 줄 구현
- [x] 미션 5
최근 본 맛집의 음식점들 Data class(식당명, 시간, 평점) → 사진도 데이터 클래스의 변수로 만들 수 있으면 포함해주세요 !
- [x] 미션 6
최근 본 맛집 음식점들 LazyComposable(LazyRow, LazyColumn) 사용해서 표현하기 (이 때 맛집 리스트의 음식점들 데이터들 겹치는 거 없게 작성해주세요 !)
- [x] 미션 7
텍스트 색상, 크기 지정 (우선은 그냥 볼드체, 색상, 크기만 해주시고, 폰트는 무시하고 진행해주세요)
- [x] 미션 8
쿠폰 받기 부분 글자색 변경 된 거 화면 회전해도 글자색 유지되도록 하기
- [x] 미션 9
화면 세로 방향으로 스크롤 가능하도록 설정(세로모드에서 정상 작동 목적)
- [x] 미션 10
전체 화면 Column으로 돼있는 거 → LazyColumn으로 변경

## 🙋 구현에 대한 설명
미션 1
서울시 광진구 구의동 부분 컴포넌트
이미지와 텍스트 컴포넌트를 이용해 구현했습니다.

미션 2
주문하러 가기 컴포넌트(주문하러 가기 부분 버튼으로 만들기)
텍스트 컴포넌트에 clickable을 적용해 구현했습니다.

미션 3
최근 본 맛집의 음식점 부분 컴포넌트
컬럼 레이아웃으로 음식 정보를 수직 배치했습니다. 

미션 4
돈가스, 일식 … 부분 LazyColumn으로 두 줄 구현
레이지 컬럼은 중첩이 되지 않기 때문에 가장 바깥의 레이지 컬럼 내부에 구현하라는 뜻으로 이해하고 컬럼 안에 넣은 형태로 레이지 컬럼 안에 넣었습니다.

미션 5
최근 본 맛집의 음식점들 Data class(식당명, 시간, 평점) → 사진도 데이터 클래스의 변수로 만들 수 있으면 포함해주세요 !
요구사항을 만족했고 평점 수도 반영했습니다.

미션 6
최근 본 맛집 음식점들 LazyComposable(LazyRow, LazyColumn) 사용해서 표현하기 (이 때 맛집 리스트의 음식점들 데이터들 겹치는 거 없게 작성해주세요 !)
마찬가지로 홈스크린 레이지컬럼 내부에 표현하라는 뜻으로 이해했고 레이지 로우로 스크롤 가능하게 배치했습니다.

미션 7
텍스트 색상, 크기 지정 (우선은 그냥 볼드체, 색상, 크기만 해주시고, 폰트는 무시하고 진행해주세요)
헥스 코드 등을 확인하여 완료했습니다.

미션 8
쿠폰 받기 부분 글자색 변경 된 거 화면 회전해도 글자색 유지되도록 하기
rememberSavable을 통해 완료했습니다.

미션 9
화면 세로 방향으로 스크롤 가능하도록 설정(세로모드에서 정상 작동 목적)
레이지 컬럼을 통해 완료했습니다.

미션 10
전체 화면 Column으로 돼있는 거 → LazyColumn으로 변경
레이지 컬럼이 중첩이 되지 않아 가장 바깥쪽에만 레이지 컬럼을 사용하고 그외 컬럼을 사용할 부분에는 일반 컬럼을 적용했습니다.

## 📷 스크린샷 & 실행영상
https://github.com/user-attachments/assets/67da605e-1186-42bc-b357-eef9b75eb723



## 🎸 기타

1hyok committed:
Feat: 홈 화면 UI 추가 (검색창, 쿠폰, 카테고리)

1hyok committed:
Feat: 카테고리 로우 추가

1hyok committed:
Feat: CityDropdown 컴포넌트 추가 및 HomeScreen에 적용

1hyok committed:
Refactor: CategoryRow LazyRow를 Row로 변경

1hyok committed:
Feat: 홈 카테고리 lazy column 제거 및 column으로 변경

1hyok committed:
Feat: MidBanner Composable 추가 및 HomeScreen에 적용

1hyok committed:
Feat: 홈 화면 최근 본 맛집 UI 구현

1hyok committed:
Feat: 최근 본 맛집 UI 구현

1hyok committed:
Feat: 중간 배너 클릭 시 로그 출력 기능 추가

중간 배너 컴포넌트(`MidBanner.kt`)에 클릭 이벤트 핸들러를 추가하여, "주문하러 가기" 버튼 클릭 시 "주문하러 가기 클릭됨!" 메시지를 콘솔에 출력하도록 수정했습니다.

`HomeScreen.kt`에서는 기존 `MidBanner` 호출 부분 아래에 주석 처리된 `MidBanner` 호출 코드가 추가되었습니다. (기능적 변경 없음)

1hyok committed:
Refactor: MidBanner 중앙 정렬 및 HomeScreen LazyColumn 적용

- `MidBanner.kt`: `MidBanner` Composable의 `fillMaxWidth`를 제거하고 `width(306.dp)`로 수정하여 중앙 정렬이 가능하도록 변경했습니다.
- `HomeScreen.kt`:
    - `LazyColumn`의 `horizontalAlignment` 주석 처리
    - `MidBanner`를 `Box`로 감싸고 `contentAlignment = Alignment.Center`를 적용하여 중앙에 배치했습니다.
    - `MidBanner` 위아래에 `Spacer`를 추가하여 간격을 조정했습니다.
- `gradle.properties`: `org.gradle.configuration-cache=true`를 추가했습니다.

1hyok committed:
Feat: NavTab enum 클래스 추가 및 BottomNavBar 리팩토링

`NavTab.kt` 파일을 새로 추가하여, 하단 네비게이션 바에 사용될 각 탭의 아이콘, 라벨, 라우트 정보를 담는 `NavTab` enum 클래스를 정의했습니다.

`MainActivity.kt`에서는 기존 `BottomNavBarItem` 리스트를 사용하는 대신, `NavTab.values()`를 사용하여 `BottomNavBar`를 구성하도록 수정했습니다. 또한, `navController.currentBackStackEntryAsState().value?.destination`을 사용하여 현재 화면의 `destination` 정보를 가져오는 로직이 추가되었으나, 아직 실제 네비게이션 상태에 따른 UI 변경은 구현되지 않았습니다. 기존 `currentIndex` 및 `navBarItems` 관련 코드는 주석 처리되었습니다.

1hyok committed:
Refactor: BottomNavBar 아이템 선택 로직 변경

- `BottomNavBar.kt`: `items`와 `currentIndex` 대신 `tabs`와 `currentTab` (NavTab enum class)을 사용하도록 수정했습니다.
- `MainActivity.kt`: `BottomNavBar` 호출 시 `NavTab`을 사용하고, 탭 선택 시 `navController.navigate` 로직을 `popUpTo`와 `launchSingleTop`을 사용하도록 변경하여 백스택 관리를 개선했습니다.

1hyok committed:
Feat: 검색 화면 및 검색 결과 화면 UI 구현, 네비게이션 로직 추가

- `SearchBar` 컴포저블을 추가하여 검색 화면과 검색 결과 화면에서 공통으로 사용합니다.
- `SearchScreen`에서 인기 검색어 목록을 표시하고, 각 항목 클릭 시 검색 결과 화면(`SearchResultScreen`)으로 이동하도록 네비게이션 로직을 구현했습니다.
- `SearchResultScreen`은 전달받은 검색어를 `SearchBar`에 표시합니다.
- 뒤로 가기 아이콘(`ic_back_arrow.xml`)을 추가했습니다.
- `MainNavHost`에 검색 관련 화면을 위한 중첩된 네비게이션 그래프(`SearchNestedGraphRoute`)를 추가하고, `SearchScreen`과 `SearchResultScreen`을 연결했습니다.

1hyok committed:
Feat: 즐겨찾기 화면 UI 구현 및 MyEats 화면에서 이동 기능 추가

- MyEats 화면에 즐겨찾기 항목 클릭 시 즐겨찾기 화면으로 이동하는 기능을 추가했습니다.
- 즐겨찾기 화면(`FavoriteScreen.kt`)에 상단 바, 총 즐겨찾기 개수 표시, 구분선 및 즐겨찾기 목록을 표시하는 UI를 구현했습니다. (목록 아이템은 아직 미구현)
- `RecentRestaurantItem.kt`에서 `RecentRestaurantData`를 `RestaurantData`로 변경하고, 관련 UI 요소를 조정했습니다. (별 아이콘 크기, 평점 텍스트 색상 등)
- `ic_back_arrow2.xml`, `like.xml` 아이콘을 추가했습니다.
- 홈 화면(`HomeScreen.kt`)에서 최근 본 맛집 제목과 목록 사이의 간격을 조정했습니다.
- `RecentRestaurantRow.kt`에서 목업 데이터를 `MockData.kt` 파일로 분리했습니다.

1hyok committed:
Refactor: 최근 본 맛집 아이템 UI 개선 및 즐겨찾기 목록에 적용

- `RecentRestaurantItem.kt` 파일명을 `RestaurantItem.kt`으로 변경하고, 내부 컴포저블 이름도 `RestaurantItem`으로 수정했습니다.
- `RestaurantItem`에서 별점 아이콘(`star.png` -> `icon_star.png`)을 교체하고, 아이콘 크기를 조정했습니다.
- `RestaurantItem`의 이미지와 텍스트 사이의 `Spacer` 높이를 `weight(1.0f)`로 변경하여 유연하게 조정되도록 수정했습니다.
- `RecentRestaurantRow.kt`에서 `RecentRestaurantItem` 대신 변경된 `RestaurantItem`을 사용하도록 수정했습니다.
- `FavoriteScreen.kt`의 즐겨찾기 목록(`LazyColumn`)에 `RestaurantItem`을 적용하고, 아이템 크기 및 정렬, 간격을 조정했습니다.

1hyok committed:
Refactor: RestaurantItem 크기 조정 로직 변경 및 적용

- `RestaurantItem.kt`: `RestaurantItem` Composable의 `modifier.size`를 주석 처리하고, 내부 `Image`에 `fillMaxWidth`를 적용하여 부모 컴포저블에서 크기를 결정하도록 변경했습니다.
- `RecentRestaurantRow.kt`: `RestaurantItem`을 호출할 때 `modifier.size`를 사용하여 `width = 231.dp`, `height = 176.dp`로 크기를 직접 지정하도록 수정했습니다.

ProtossManse commented on app/src/main/java/com/kuit/kuit6android/ui/home/component/CategoryRow.kt:
```diff
@@ -0,0 +1,39 @@
+    )
+//    LazyRow(horizontalArrangement = Arrangement.spacedBy(space = 4.dp)) {
+//    LazyRow(horizontalArrangement = Arrangement.spacedBy(space = 18.dp)) {
+    Row(horizontalArrangement = Arrangement.spacedBy(space = 18.dp)) {
```
```suggestion
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 27.dp),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
```
spacedBy로만 여백을 주면 기기가 가로로 길어졌을 때, 좌우 여백이 너무 클 수 있기 때문에 FillMaxWidth와 SpaceBetween 사용하는 게 더 좋을 것 같습니다!

ProtossManse commented on app/src/main/java/com/kuit/kuit6android/ui/home/component/MidBanner.kt:
```diff
@@ -0,0 +1,98 @@
+//                width = 306.dp,
+//                height = 77.dp
+//            )
+            .width(306.dp)
```
```suggestion
            .fillMaxWidth()
            .padding(horizontal = 27.dp)
```

이 부분도 상기한 이유와 같은 이유로 고정 폭 대신 화면 전체 너비를 채우고, 패딩을 주는 방식이 더 좋을 것 같습니다 !

ProtossManse reviewed (approved):
말씀드린 부분 외에는 잘 구현하셨습니다 ! 
리뷰 내용대로 수정하면 아래와 같이 되니 참고 부탁드려요
늦어서 죄송합니다,,ㅎㅎ 수고하셨어요!

Before
<img width="363" height="543" alt="Image" src="https://github.com/user-attachments/assets/7892e03c-e50b-438a-8c7b-a8e716dfc6ec" />

After
<img width="375" height="618" alt="Image" src="https://github.com/user-attachments/assets/fc9adfbd-4501-4658-ab79-4b4de162f419" />

1hyok commented:
> 말씀드린 부분 외에는 잘 구현하셨습니다 ! 리뷰 내용대로 수정하면 아래와 같이 되니 참고 부탁드려요 늦어서 죄송합니다,,ㅎㅎ 수고하셨어요!
> 
> Before <img alt="Image" width="363" height="543" src="https://private-user-images.githubusercontent.com/85992666/499314171-7892e03c-e50b-438a-8c7b-a8e716dfc6ec.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjAxMDc2NDEsIm5iZiI6MTc2MDEwNzM0MSwicGF0aCI6Ii84NTk5MjY2Ni80OTkzMTQxNzEtNzg5MmUwM2MtZTUwYi00MzhhLThjN2ItYThlNzE2ZGZjNmVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDEwVDE0NDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhYjVjMWVjNmJiOTM1ZGNjNzBkMGNhZWZlY2IyZjA4NWQ3N2FmMGEyMWEzZTg5YzY3ZGVkNmU4NzEzZmEyODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c-lV-9fRuG9QavlY4vzVVQyTHkkkM1ifVkkC4GAQxu4">
> 
> After <img alt="Image" width="375" height="618" src="https://private-user-images.githubusercontent.com/85992666/499314399-fc9adfbd-4501-4658-ab79-4b4de162f419.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjAxMDc2NDEsIm5iZiI6MTc2MDEwNzM0MSwicGF0aCI6Ii84NTk5MjY2Ni80OTkzMTQzOTktZmM5YWRmYmQtNDUwMS00NjU4LWFiNzktNGI0ZGUxNjJmNDE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDEwVDE0NDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNWI1YmRkOWQ1NzMxNTEyMGJkNjAwNWE3M2VjMzViZTYwNDRmYzZhY2U2NmRlYjlmNDg5YTRlZWE0NmE3MmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ur9Pqz3eXXUHFyOX7fDDVBYnuIJ6s5hV_1FJUbSX_c4">



> 말씀드린 부분 외에는 잘 구현하셨습니다 ! 리뷰 내용대로 수정하면 아래와 같이 되니 참고 부탁드려요 늦어서 죄송합니다,,ㅎㅎ 수고하셨어요!
> 
> Before <img alt="Image" width="363" height="543" src="https://private-user-images.githubusercontent.com/85992666/499314171-7892e03c-e50b-438a-8c7b-a8e716dfc6ec.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjAxMDc2NDEsIm5iZiI6MTc2MDEwNzM0MSwicGF0aCI6Ii84NTk5MjY2Ni80OTkzMTQxNzEtNzg5MmUwM2MtZTUwYi00MzhhLThjN2ItYThlNzE2ZGZjNmVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDEwVDE0NDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhYjVjMWVjNmJiOTM1ZGNjNzBkMGNhZWZlY2IyZjA4NWQ3N2FmMGEyMWEzZTg5YzY3ZGVkNmU4NzEzZmEyODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c-lV-9fRuG9QavlY4vzVVQyTHkkkM1ifVkkC4GAQxu4">
> 
> After <img alt="Image" width="375" height="618" src="https://private-user-images.githubusercontent.com/85992666/499314399-fc9adfbd-4501-4658-ab79-4b4de162f419.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjAxMDc2NDEsIm5iZiI6MTc2MDEwNzM0MSwicGF0aCI6Ii84NTk5MjY2Ni80OTkzMTQzOTktZmM5YWRmYmQtNDUwMS00NjU4LWFiNzktNGI0ZGUxNjJmNDE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDEwVDE0NDIyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjNWI1YmRkOWQ1NzMxNTEyMGJkNjAwNWE3M2VjMzViZTYwNDRmYzZhY2U2NmRlYjlmNDg5YTRlZWE0NmE3MmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ur9Pqz3eXXUHFyOX7fDDVBYnuIJ6s5hV_1FJUbSX_c4">

감사합니다 수고하셨어용!

1hyok committed:
Chore: Gradle 및 Kotlin 버전 업데이트와 빌드 환경 설정 추가

Gradle과 Kotlin의 버전을 최신으로 업데이트하고, `gradle.properties`에 안드로이드 빌드 최적화 및 호환성을 위한 다양한 설정값들을 추가했습니다.

- **버전 업데이트 (`libs.versions.toml`, `gradle-wrapper.properties`)**
    - Android Gradle Plugin(AGP) 버전을 `8.13.0`에서 `9.1.0`으로 업데이트했습니다.
    - Kotlin 버전을 `2.0.21`에서 `2.2.10`으로 업데이트했습니다.
    - Gradle Wrapper 버전을 `8.13`에서 `9.3.1`로 업데이트했습니다.

- **빌드 환경 설정 (`gradle.properties`)**
    - `android.defaults.buildfeatures.resvalues`, `android.dependency.useConstraints` 등 빌드 기능 및 의존성 관리에 필요한 여러 옵션들을 추가했습니다.
    - R8 리소스 최적화 및 엄격 모드 관련 설정을 추가했습니다.

- **기타**
    - `GetCouponItem.kt`: 코드 스타일 가이드에 맞춰 불필요한 공백을 제거하고 콤마(,)를 추가했습니다.
    - 불필요한 Kotlin 컴파일러 세션 파일(`.salive`)을 삭제했습니다.
